core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -84,8 +84,14 @@ public T index(boolean index) {
                      * if the fieldType has a non-null option we are all good it might have been set through a different
                      * call.
                      */
-                    final IndexOptions options = getDefaultIndexOption();
-                    assert options != IndexOptions.NONE : "default IndexOptions is NONE can't enable indexing";
+                    IndexOptions options = getDefaultIndexOption();
+                    if (options == IndexOptions.NONE) {
+                        // this can happen if a metadata field was configured
+                        // to be not indexed on another type
+                        // so we use IndexOptions.DOCS here but this will result
+                        // in a mapping merge later
+                        options = IndexOptions.DOCS;
```

Comment:
I read the comment...but I still dont' understand...why can't we just throw an error?